diff --git a/handlers/helper.go b/handlers/helper.go
--- a/handlers/helper.go
+++ b/handlers/helper.go
@@ -25,6 +25,7 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
 	"github.com/civo/civogo"
 	"github.com/digitalocean/godo"
 	"github.com/linode/linodego"
@@ -68,7 +69,14 @@ func makeClientFromAccount(account models.Account) (*providers.ProviderClient, e
 		}
 	} else {
-			cfg, err := awsConfig.LoadDefaultConfig(context.Background())
+			cfg, err := awsConfig.LoadDefaultConfig(
+				context.Background(),
+				awsConfig.WithCredentialsProvider(
+					credentials.NewStaticCredentialsProvider(
+						account.Credentials["aws_access_key_id"],
+						account.Credentials["aws_secret_access_key"],
+						"",
+					),
+				),
+			)
 			if err != nil {
 				return nil, err
 			}
